### PR TITLE
Use correct solcjs binary for testing.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by axic** [comment]
If we can have a reliable check that the project uses truffle >5.0 then we can add a single line to `truffle.config` so point it to a specific solc version. No messing around needed.

Documentation: https://truffleframework.com/docs/truffle/reference/configuration#solc

**Comment by chriseth** [comment]
Isn't that what this does?

**Review by jjlk40 — COMMENTED** [review]
No change

**Comment by axic** [comment]
> Isn't that what this does?

Right, you're correct, however it replaces the config and that could break projects if they rely on certain settings for ganache.

**Comment by chriseth** [comment]
It does not replace the config, it appends. It might overwrite optimizer settings, though.

**Comment by axic** [comment]
Oh, correct, you may want to override the entire `solc` section though.

File: test/externalTests.sh [hunk 1]
```diff
@@ -44,45 +44,42 @@ function test_truffle
...
       fi
-      assertsol="node_modules/truffle/build/Assert.sol"
-      if [ -f "$assertsol" ]
-      then
-        echo "Replace Truffle's Assert.sol with a known good version"
-        rm "$assertsol"
-        wget https://raw.githubusercontent.com/trufflesuite/truffle-core/ef31bcaa15dbd9bd0f6a0070a5c63f271cde2dbc/lib/testing/Assert.sol -o "$assertsol"
-      fi
       # Change "compileStandard" to "compile"
       sed -i s/solc.compileStandard/solc.compile/ "node_modules/truffle/build/cli.bundled.js"
```
Comment (axic) [inline test/externalTests.sh:88]: Since below we assume Truffle 5.x, these two lines can be also removed.

File: test/externalTests.sh [hunk 2]
```diff
@@ -44,45 +44,42 @@ function test_truffle
...
+      cat >> truffle*.js <<EOF
+module.exports['compilers']['solc'] = {version: "$DIR/solc"};
```
Comment (axic) [inline test/externalTests.sh]: Can you leave a comment here that if no `evmVersion` is set, Truffle defaults to Byzantium. We may want to override that with Constantinople here when #6084 is turned on.

File: test/externalTests.sh [hunk 3]
```diff
@@ -44,45 +44,42 @@ function test_truffle
...
-      assertsol="node_modules/truffle/build/Assert.sol"
-      if [ -f "$assertsol" ]
-      then
-        echo "Replace Truffle's Assert.sol with a known good version"
-        rm "$assertsol"
-        wget https://raw.githubusercontent.com/trufflesuite/truffle-core/ef31bcaa15dbd9bd0f6a0070a5c63f271cde2dbc/lib/testing/Assert.sol -o "$assertsol"
-      fi
       # Change "compileStandard" to "compile"
       sed -i s/solc.compileStandard/solc.compile/ "node_modules/truffle/build/cli.bundled.js"
+      cat >> truffle*.js <<EOF
```
Comment (axic) [inline test/externalTests.sh:90]: Also please add a comment above this line stating we require Truffle 5.x.

There's no need to add a runtime check to verify it is Truffle 5.x until we run into projects which don't use it.

**Comment by chriseth** [comment]
Hm, it seems like safe-contracts is still using truffle 4.1.14

**Comment by chriseth** [comment]
Trying a hybrid approach....

**Comment by chriseth** [comment]
Wohoo!

**Review by erak — APPROVED** [review]
This solution does not work if the project contrains multiple truffle config files e.g. https://github.com/PolymathNetwork/polymath-core/tree/dev-3.0.0. But I think we can merge it now and take care of the other aspect later.

File: test/externalTests.sh [hunk 4]
```diff
@@ -73,16 +84,16 @@ function test_truffle
         # Replace fixed-version pragmas in Gnosis (part of Consensys best practice)
         find contracts test -name '*.sol' -type f -print0 | xargs -0 sed -i -e 's/pragma solidity [\^0-9\.]*/pragma solidity >=0.0/'
       fi
-      assertsol="node_modules/truffle/build/Assert.sol"
-      if [ -f "$assertsol" ]
-      then
-        echo "Replace Truffle's Assert.sol with a known good version"
-        rm "$assertsol"
-        wget https://raw.githubusercontent.com/trufflesuite/truffle-core/ef31bcaa15dbd9bd0f6a0070a5c63f271cde2dbc/lib/testing/Assert.sol -o "$assertsol"
```
Comment (axic) [inline test/externalTests.sh:81]: I think this should have stayed if we still support Truffle 4.x

Comment (chriseth) [reply test/externalTests.sh:81]: They are identical, I checked.

Comment (axic) [reply test/externalTests.sh:81]: Cool! Depends on the truffle version though. We'll see if something breaks in the future anyway.

**Pull request merged**